pkgs/by-name/wi/wivrn/package.nix

```diff
@@ -0,0 +1,123 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, vulkan-headers
+, vulkan-loader
+, x264
+, xorg
+, eigen
+, boost
+, glm
+, spdlog
+, monado
+, openxr-loader
+, nlohmann_json
+, harfbuzz
+, freetype
+, glslang
+, python3
+, udev
+, ffmpeg
+, libva
+, libdrm
+, avahi
+, libpulseaudio
+, pkgs
+}:
+let
+  # We need tiny_gltf.h and draco doesn't provide it
+  tinygltf = pkgs.callPackage "${pkgs.path}/pkgs/development/libraries/draco/tinygltf.nix" {};
+
+  # The commit of monado specified in CMakeLists.txt
+  vendorMonado = monado.overrideAttrs rec {
+    postInstall = ''
+      mv src/xrt/compositor/libcomp_main.a $out/lib/libcomp_main.a
+    '';
+    version = "79bf8eb8fa168f65f4e5505e0525ee74aa88783e";
+    src = pkgs.fetchgit {
+      url = "https://gitlab.freedesktop.org/monado/monado.git";
+      rev = version;
+      hash = "sha256-lzvu8xw/10pMP4pWzEDYKKoBL6GbzsmbeOCqLsuN2lk=";
+    };
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "wivrn";
+  version = "0.10.1";
+
+  CPM_LOCAL_PACKAGES_ONLY = "ON";
```

Comment:
Maybe move closer to `cmakeFlags` because this is effective during the configuration phase?